packages/spotlight/src/vite-plugin.ts

```diff
@@ -0,0 +1,293 @@
+          const sourceMapContent = Buffer.from(sourceMapBase64, 'base64').toString('utf-8');
+          await applySourceContextToFrame(sourceMapContent, frame);
+        }
+      }
+    }
+
+    const responseJson = JSON.stringify(stacktrace);
+
+    res.writeHead(200, { 'Content-Type': 'application/json' });
+    res.end(responseJson);
+  });
+};
+
+async function sendErrorToSpotlight(err: ErrorPayload['err'], spotlightUrl: string = 'http://localhost:8969/stream') {
+  const error = err.errors[0];
+  const contextLines = err.pluginCode?.split('\n');
+  const errorLine = error.location.lineText;
+  const errorLineInContext = contextLines?.indexOf(errorLine);
+  const event_id = randomBytes(16).toString('hex');
+  const timestamp = new Date();
+  const envelope = [
+    { event_id, sent_at: timestamp.toISOString() },
+    { type: 'event' },
+    {
+      event_id,
+      level: 'error',
+      platform: 'javascript',
+      environment: 'development',
+      tags: { runtime: 'vite' },
+      timestamp: timestamp.getTime(),
+      exception: {
+        values: [
+          {
+            type: 'Error',
+            mechanism: {
+              type: 'instrument',
+              handled: false,
+            },
+            value: error.text,
+            stacktrace: {
+              frames: [
+                error
+                  ? {
+                      filename: error.location.file,
+                      lineno: error.location.line,
+                      colno: error.location.column,
+                      context_line: errorLine,
+                      pre_context: contextLines?.slice(0, errorLineInContext),
+                      post_context:
+                        errorLineInContext != null && errorLineInContext > -1
+                          ? contextLines?.slice(errorLineInContext + 1)
+                          : undefined,
+                    }
+                  : {
+                      filename: err.id,
+                    },
+              ],
+            },
+          },
+        ],
+      },
+    },
+  ]
+    .map(p => JSON.stringify(p))
+    .join('\n');
+  return await fetch(spotlightUrl, {
+    method: 'POST',
+    body: envelope,
+    headers: { 'Content-Type': 'application/x-sentry-envelope' },
+  });
+}
+
+export default function spotlight({ port }: { port?: number } = {}): PluginOption {
+  let spotlightPath: string;
+
+  return {
+    name: 'spotlight',
+    apply: 'serve',
+    transform(code: string, id: string /*, _opts = {}*/) {
+      // Only modify Vite's special client script
+      // The magic value below comes from vite/constants:CLIENT_ENTRY
+      if (!id.endsWith('vite/dist/client/client.mjs')) return;
+
+      const initSnippet = [
+        'createErrorOverlay=()=>{}',
+        `import * as Spotlight from '/@fs${spotlightPath}'`,
+        'Spotlight.init({injectImmediately: true, integrations: [Spotlight.sentry({openLastError: enableOverlay})]})',
```

Comment:
Hi @BYK correct me if i m wrong here.
this code snippet will be responsible of injecting spotlight overlay?
if yes, then ideally we should allow user to pass any overlay init options, right?

And, for SSR errors, we have the logic in spotlight/astro to display fullpage spotlight.

Reply:
>if yes, then ideally we should allow user to pass any overlay init options, right?

Yes it is. I was torn between providing options or just making it a straightforward, seamless experience. Is there an option you think we definitely should allow changing here?

>And, for SSR errors, we have the logic in spotlight/astro to display fullpage spotlight.

Yup, I'll check that out good pointer.

Reply:
> Yes it is. I was torn between providing options or just making it a straightforward, seamless experience. Is there an option you think we definitely should allow changing here?

I think we can give every option, because projects using this vite plugin, will rely on the config here only for spotlight.